### ISSUE TITLE
[FEATURE] Parse annotations from the SANY comments

### ISSUE THREAD

**Issue by konnov** [informalsystems/apalache#226]
As discussed in #162, it should be possible to write type annotations in comments:

```tla
VARIABLE
  \* type: Set(Int)
  S

\* type: (Int) => Bool
Mem(e) == \E x \in S: x = e
```

We should integrate annotations in `SanyImporter` and propagate them to the type checker.

**Comment by konnov** [comment]
It is actually a good idea to hide the annotations in comments. Our current approach with extending the Apalache module introduces an additional dependency.

**Comment by konnov** [comment]
This is implemented in #503.
